docs/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,141 @@
+`pytest-benchmark` has a number of additional options that can be used to customize the output.
+The next line contains one useful example, but developers should experiment to find a useful output
+```
+pytest-benchmark compare XXXX YYYY --sort="name" --columns=Mean --name=short --group-by=param
+```
+
+For more details, see the [`pytest-benchmark` documentation](https://pytest-benchmark.readthedocs.io/en/latest/comparing.html).
+
+## Benchmark contents
+
+### Benchmark configuration
+
+Benchmarks must support [comparing to geopandas](#comparing-to-geopandas) and [being run as tests](#testing-benchmarks).
+To satisfy these requirements, one must follow these rules when writing benchmarks:
+1. Import `cuspatial` and `cupy` from the config module:
+   ```python
+       from ..common.config import cuspatial  # Do this
+       import cuspatial  # Not this
+   ```
+   This enables swapping out for `geopandas`.
+2. Avoid hard-coding benchmark dataset sizes, and instead use the sizes advertised by `config.py`.
+   This enables running the benchmarks in "test" mode on small datasets, which will be much faster.
+
+
+### Writing benchmarks 
+
+Just as benchmarks should be written in terms of the highest level classes in the hierarchy,
+they should also assume as little as possible about the nature of the data.
+
+
+## Comparing to geopandas
+
+An important aspect of benchmarking cuSpatial is comparing it to geopandas.
+We often want to generate quantitative comparisons, so we need to make that as easy as possible.
+Our benchmarks support this by setting the environment variable `CUSPATIAL_BENCHMARKS_USE_GEOPANDAS`.
+When this variable is detected, all benchmarks will automatically be run using geopandas instead of cuspatial.
+Therefore, comparisons can easily be generated by simply running the benchmarks twice,
+once with the variable set and once without.
+Note that this variable only affects API benchmarks, not internal benchmarks,
+since the latter are not even guaranteed to be valid geopandas code.
+
+```{note}
+`CUSPATIAL_BENCHMARKS_USE_GEOPANDAS` effectively remaps `cuspatial` to `geopandas`.
+It does so by aliasing these modules in `common.config.py`.
+This aliasing is why it is critical for developers to import these packages from `config.py`.
+```
+
+## Testing benchmarks
+
+Benchmarks need to be kept up to date with API changes in cuspatial.
+However, we cannot simply run benchmarks in CI.
+Doing so would consume too many resources, and it would significantly slow down the development cycle
+
+To balance these issues, our benchmarks also support running in "testing" mode.
+To do so, developers can set the `CUSPATIAL_BENCHMARKS_DEBUG_ONLY` environment variable.
+When benchmarks are run with this variable, all data sizes are set to a minimum and the number of sizes are reduced.
+Our CI testing takes advantage of this to ensure that benchmarks remain valid code.
```

Comment:
Also unique to cuDF. Not sure if we have a (simple) alternative to the problem raised here without introducing cuDF's benchmark suite.

Reply:
I dropped the cuDF unique sections or modified them considerably.